ecl/hql/hqlmanifest.cpp

```diff
@@ -198,6 +198,52 @@ void ResourceManifest::loadResource(const char *filepath, MemoryBuffer &content)
     read(fio, 0, (size32_t) f->size(), content);
 }
 
+void callJfrog(StringBuffer &localpath, const char *manifestFilename, IPropertyTree &item)
+{
+    const char *repo = item.queryProp("@repository");
+    if(repo)
+    {
+        StringBuffer downloadPath("/tmp/HPCCSystems/jfrog_downloads/");
+        Owned<IFile> dir = createIFile(downloadPath.str());
+        if(!dir->createDirectory())
+            throw makeStringExceptionV(0,"Error creating download directory: %s", downloadPath.str());
+        StringBuffer filename(localpath);
+        getFileNameOnly(filename, false);
+        downloadPath.append(filename);
+        DBGLOG("jfrog dl path: %s", downloadPath.str());
+        StringBuffer filepath;
+        StringBuffer machine;
+        if (repo[0] == '/' && repo[1] == '/')
+            splitUNCFilename(repo, &machine, &filepath, nullptr, nullptr);
+        else
+            splitFilename(repo, nullptr, &filepath, nullptr, nullptr);
+        StringBuffer jfrog_cmd("jf rt dl --flat --quiet ");
+        if (!machine.isEmpty())
+            jfrog_cmd.appendf("--url=http:%s/artifactory ", machine.str());
+        StringBuffer userValue;
+        StringBuffer keyValue;
+        const char *repoName = item.queryProp("@repoName");
+        if (repoName && getSecretValue(keyValue, "jfrog", repoName, "key", true) && getSecretValue(userValue, "jfrog", repoName, "user", true))
```

Comment:
Is there a more secure way of passing the credentials?  Command lines can be read by other processes.

Reply:
The user can do a one time configuration of the artifactory connection to avoid passing credentials each time we download from JFrog. I removed the code for passing credentials via the command line.